src/error.rs

```diff
@@ -7,8 +8,18 @@ pub enum ODataError {
     #[error(transparent)]
     UnsupportedDataType(#[from] UnsupportedDataType),
     #[error(transparent)]
+    UnsupportedColumnType(#[from] UnsupportedColumnType),
+    #[error(transparent)]
+    BatchUnexpectedRowsNumber(#[from] BatchUnexpectedRowsNumber),
+    #[error(transparent)]
+    UnexpectedBatchesNumber(#[from] UnexpectedBatchesNumber),
```

Comment:
Don't think these errors belong here. `ODataError` is an API consumer facing type, but these two errors are thrown only when data provider incorrectly implements the query. Thus I think they don't need to be present in the enum and can be wrapped directly into `ODataError::internal()`